test/regression/RegressionTest.cc

```diff
@@ -37,8 +37,12 @@ void RegressionTest::run(const char* source, const char* triple,
   // Initialise the simulation memory
   memory_ = std::make_shared<simeng::memory::SimpleMem>(memorySize);
 
-  // Initialise a SimOS object.
+  // Initialise a SimOS object and extract callback function for communicating a
+  // syscall to the Syscall Handler
   simeng::OS::SimOS OS = simeng::OS::SimOS(DEFAULT_STR, {}, memory_, true);
+  simeng::sendSyscallToHandler syscallHandle = [&](auto syscallInfo) {
+    return OS.receiveSyscall(syscallInfo);
+  };
```

Comment:
is `[&]` the correct scope to pass?

Reply:
This should be refactored to `OS.getSyscallReceiver()`

Reply:
Yes the scope in which resides OS would be correct to pass, otherwise if [OS] is passed, the lambda will try to copy it and fail because OS cannot be copied as it contains a unique_ptr.
Giving out this error:
```
/Users/jj16791/Documents/GitHub/SimEng/test/regression/RegressionTest.cc:44:8: error: call to implicitly-deleted copy constructor of 'simeng::OS::SimOS'
      [OS](auto syscallInfo) { return OS.receiveSyscall(syscallInfo); };
       ^~
/Users/jj16791/Documents/GitHub/SimEng/src/include/simeng/OS/SimOS.hh:86:46: note: copy constructor of 'SimOS' is implicitly deleted because field 'hfmmap_' has a deleted copy constructor
  std::unique_ptr<HostBackedFileMMaps> const hfmmap_ =
```